core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala

```diff
@@ -371,6 +355,156 @@ object RawFeatureFilter {
     bins
   }
 
+  private def getAllSummaries(bins: Int, allFeatures: RDD[AllFeatures]): AllSummaries = {
+    def apply(sum: AllSummaries, feat: AllFeatures): AllSummaries = {
+      val (totalCount, responseSummaries, numericSummaries, textSummaries) = sum
+      val (responseFeatures, numericFeatures, textFeatures) = feat
+
+      def updateNumericSummaries(
```

Comment:
this might be just my personal feelings, is this nested too much? e.g. three levels nested method